File: mobile-client/android/UseCase/System/Config/RefreshRemoteConfigUseCase.kt
package app.mobiling.client.usecase.system.config

import app.mobiling.client.data.system.config.RemoteConfigStore

class RefreshRemoteConfigUseCase(private val remoteConfigStore: RemoteConfigStore) {
    suspend operator fun invoke(
        baseUrl: String = "https://httpbin.org",
        queueMaxAttempt: Int = 3,
        backoffBaseMs: Int = 500,
    ): Boolean = remoteConfigStore.refresh(
        baseUrl = baseUrl,
        queueMaxAttempt = queueMaxAttempt,
        backoffBaseMs = backoffBaseMs,
    )
}

File: client/ios/UseCase/System/Config/RefreshRemoteConfigUseCase.swift
import Foundation

public struct RefreshRemoteConfigUseCase {
    private let remoteConfigStore: RemoteConfigStore

    public init(remoteConfigStore: RemoteConfigStore) {
        self.remoteConfigStore = remoteConfigStore
    }

    public func callAsFunction(baseUrl: String = "https://httpbin.org") async throws -> Bool {
        try await remoteConfigStore.refresh(baseUrl: baseUrl)
    }
}
